src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/AbstractImplicitlyActiveComponent.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.ProjectSystem
+{
+    /// <summary>
+    ///     An <see langword="abstract"/> base class that loads, or unloads its inner instance when a 
+    ///     <see cref="ConfiguredProject"/> becomes implicitly activated, or deactivated, respectively.
+    /// </summary>
+    internal abstract class AbstractImplicitlyActiveComponent : AbstractProjectDynamicLoadComponent
+    {
+        private readonly IConfiguredProjectImplicitActivationTracking _activationTracking;
+
+        protected AbstractImplicitlyActiveComponent(IConfiguredProjectImplicitActivationTracking activationTracking, 
+                                                 JoinableTaskContextNode joinableTaskContextNode)
+            : base(joinableTaskContextNode)
+        {
+            Requires.NotNull(activationTracking, nameof(activationTracking));
+
+            _activationTracking = activationTracking;
+        }
+
+        protected override Task InitializeCoreAsync(CancellationToken cancellationToken)
+        {
+            _activationTracking.ImplicitlyActivated += OnImplicitlyActivated;
+            _activationTracking.ImplicitlyDeactivated += OnImplicitlyDeactivated;
+
+            if (_activationTracking.IsImplicitlyActive)
+                return LoadAsync();
```

Comment:
Is `LoadAsync` meant to be idempotent?  Isn't there a race here if the event fires between subscription and here?

Reply:
Yes it is.

Reply:
There is a race here; if we deactivate after we check this property - we'll Load but never unload until project tear down.

Reply:
Okay, I've figured out a better way to do this - I'm going to drop the events.

Reply:
https://github.com/dotnet/project-system/pull/3689